From 24dc68037a01e0f73acea0dec8fd918837efe919 Mon Sep 17 00:00:00 2001
Message-Id: <24dc68037a01e0f73acea0dec8fd918837efe919.1367947969.git.minovotn@redhat.com>
In-Reply-To: <707b9b97153063374d2530e72c49b1499fc21af9.1367947969.git.minovotn@redhat.com>
References: <707b9b97153063374d2530e72c49b1499fc21af9.1367947969.git.minovotn@redhat.com>
From: Laszlo Ersek <lersek@redhat.com>
Date: Mon, 6 May 2013 19:27:57 +0200
Subject: [PATCH 092/114] qemu-ga: Plug fd leak on ga_channel_listen_accept()
 error path

RH-Author: Laszlo Ersek <lersek@redhat.com>
Message-id: <1367868499-27603-35-git-send-email-lersek@redhat.com>
Patchwork-id: 51133
O-Subject: [RHEL-6.5 qemu-kvm PATCH v2 34/56] qemu-ga: Plug fd leak on ga_channel_listen_accept() error path
Bugzilla: 952873
RH-Acked-by: Jeffrey Cody <jcody@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

From: Markus Armbruster <armbru@redhat.com>

Spotted by Coverity.

Signed-off-by: Markus Armbruster <armbru@redhat.com>
Reviewed-by: Eric Blake <eblake@redhat.com>
Reviewed-by: Michael Roth <mdroth@linux.vnet.ibm.com>
Reviewed-by: Luiz Capitulino <lcapitulino@redhat.com>
Signed-off-by: Michael Roth <mdroth@linux.vnet.ibm.com>
(cherry picked from commit 32c16620dda8ba16f6d6bcd20efefdec8975af77)
Signed-off-by: Laszlo Ersek <lersek@redhat.com>
---
 qga/channel-posix.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 qga/channel-posix.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/qga/channel-posix.c b/qga/channel-posix.c
index 52f880d..2a1fe03 100644
--- a/qga/channel-posix.c
+++ b/qga/channel-posix.c
@@ -46,6 +46,7 @@ static gboolean ga_channel_listen_accept(GIOChannel *channel,
     ret = ga_channel_client_add(c, client_fd);
     if (ret) {
         g_warning("error setting up connection");
+        close(client_fd);
         goto out;
     }
     accepted = true;
-- 
1.7.11.7

